src/prefect/utilities/callables.py

```diff
@@ -47,6 +47,82 @@ def get_call_parameters(
     return dict(bound_signature.arguments)
 
 
+def explode_variadic_parameter(
+    fn: Callable, parameters: Dict[str, Any]
+) -> Dict[str, Any]:
+    """
+    Given a parameter dictionary, move any parameters stored in a variadic keyword
+    argument parameter (i.e. **kwargs) into the top level.
+
+    Example:
+
+        ```python
+        def foo(a, b, **kwargs):
+            pass
+
+        parameters = {"a": 1, "b": 2, "kwargs": {"c": 3, "d": 4}}
+        explode_variadic_parameter(foo, parameters)
+        # {"a": 1, "b": 2, "c": 3, "d": 4}
+        ```
+    """
+    variadic_key = None
+    for key, parameter in inspect.signature(fn).parameters.items():
+        if parameter.kind == parameter.VAR_KEYWORD:
+            variadic_key = key
+            break
+
+    if not variadic_key:
+        return parameters
+
+    new_parameters = parameters.copy()
+    for key, value in new_parameters.pop(variadic_key).items():
+        new_parameters[key] = value
+
+    return new_parameters
+
+
+def collapse_variadic_parameters(
+    fn: Callable, parameters: Dict[str, Any]
+) -> Dict[str, Any]:
+    """
+    Given a parameter dictionary, move any parameters stored not present in the
+    signature into the variadic keyword argument.
+
+    Example:
+
+        ```python
+        def foo(a, b, **kwargs):
+            pass
+
+        parameters = {"a": 1, "b": 2, "c": 3, "d": 4}
+        collapse_variadic_parameters(foo, parameters)
+        # {"a": 1, "b": 2, "kwargs": {"c": 3, "d": 4}}
+        ```
+    """
+    signature_parameters = inspect.signature(fn).parameters
+    variadic_key = None
+    for key, parameter in signature_parameters.items():
+        if parameter.kind == parameter.VAR_KEYWORD:
```

Comment:
I'm not sure whether `*args` works correctly at the moment, but if so, this function could also look for `parameter.kind == parameter.VAR_POSITIONAL`.

Reply:
I was thinking something similar. I presume the same issue will occur with `*args` but I'd rather tackle that after we've agreed on a design for this.

Thanks for the review!